File: Kattis/challenge/ranklist.java
import java.util.*;


public class ranklist
{
    public static void main(String[] args)
    {
        Scanner in = new Scanner(System.in);
        int numCases = in.nextInt();

        while(numCases-- > 0)
        {
            int numTeams = in.nextInt();
            
            int[] teams = new int[numTeams];
            for(int i = 0; i < numTeams; i++) {
                in.next();
                teams[i] = in.nextInt();
            }
            
            Arrays.sort(teams);
            
            int diff = 0;
            for(int i = 0; i < teams.length; i++)
                diff += Math.abs((i+1) - teams[i]);
            
            System.out.println(diff);
        }
    }
}

File: Kattis/challenge/ranklist.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int numCase;
    cin >> numCase;
    
    while (numCase-- > 0)
    {
        cin.ignore();
        int numTeams;
        cin >> numTeams;

        vector<int> teams(numTeams);
        for(int i = 0; i < numTeams; i++) {
            string name;
            cin >> name >> teams[i];
        }

        sort(teams.begin(), teams.end());

        int diff = 0;
        for(int i = 0; i < teams.size(); i++)
            diff += abs((i+1) - teams[i]);
        
        cout << diff << endl;
    }
    
    
}